src/lib/expressionBuilder.ts

```diff
@@ -151,10 +151,13 @@ const parseClause = (_clause: FilterExpression, grp: number, table: Table) => {
   if (entity !== undefined && (typeof entity !== 'string' || (!table[entity] || !table[entity].schema || !table[entity].schema.attributes)))
     error(`'entity' value of '${entity}' must be a string and a valid table Entity name`)
 
-  // Add filter attribute to names
-  names[`#attr${grp}`] = typeof attr === 'string' ? checkAttribute(attr,(entity ? table[entity].schema.attributes : table.Table.attributes))
-    : typeof size === 'string' ? checkAttribute(size,(entity ? table[entity].schema.attributes : table.Table.attributes))
-    : error(`A string for 'attr' or 'size' is required for condition expressions`)
+  // Add filter attribute to names  
+  let attrName = "";
+  const path = typeof attr === 'string' ? attr.split('.') : typeof size === 'string' ? size.split('.') : error(`A string for 'attr' or 'size' is required for condition expressions`);;
+  for (let i = 0; i < path.length; i++) {
+    names[i == 0 ? `#attr${grp}` : `#attr${grp}${i}`] = i == 0 ? checkAttribute(path[i], (entity ? table[entity].schema.attributes : table.Table.attributes)) : path[i];
```

Comment:
I think this would be easier to understand if you used `if`...`else` statements rather than inlining it like this. Took me a while to read it and process how it's supposed to work.